rockcraft/oci.py

```diff
@@ -365,24 +366,33 @@ def set_entrypoint(self, entrypoint_service: Optional[str]) -> None:
         params = ["--clear=config.entrypoint"]
         for entry in entrypoint:
             params.extend(["--config.entrypoint", entry])
+        params.extend(["--clear=config.cmd"])
         _config_image(image_path, params)
         emit.progress(f"Entrypoint set to {entrypoint}")
 
     def set_cmd(self, command: Optional[str]) -> None:
         """Set the OCI image CMD."""
         emit.progress("Configuring CMD...")
         image_path = self.path / self.image_name
-        cmd_params = ["--clear=config.cmd"]
+        cmd_params = []
```

Comment:
it was moved to set_entrypoint. otherwise if we keep it here we will have to call set_cmd in all cases. we've disccused it before iirc and you opted to have clear=config.cmd back in set_entrypoint.

Reply:
yes it should be in `set_entrypoint`. But what I meant is that it should also be in this fn. I get that right now, `cmd` has already been cleared when we get to this fn, and thus this might feel redundant. It's just that this is enforced by design. So I'm afraid that if tomorrow we change the ROCK's design such that one can set the CMD without setting the entrypoint, then this would be wrong.